modules/nw-http-header-configuration.adoc

```diff
@@ -0,0 +1,135 @@
+
+.Example Ingress Controller spec
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+...
+spec:
+  httpHeaders:
+    actions:
+      response:
+      - name: X-Frame-Options
+        action:
+          type: Set
+          set:
+            value: DENY
+----
+
+And if a route owner sets the same X-Frame-Options response header with the value `SAMEORIGIN`:
+
+.Example route spec
+[source,yaml]
+----
+apiVersion: route.openshift.io/v1
+kind: Route
+...
+spec:
+  httpHeaders:
+    actions:
+      response:
+      - name: X-Frame-Options
+        action:
+          type: Set
+          set:
+            value: SAMEORIGIN
+----
+
+Then the value set for the X-Frame-Options header at the global level in the Ingress Controller will take precedence, even if a specific route allows frames.
+
+This is because `haproxy.config` uses the following logic, where the Ingress Controller is considered the frontend and individual routes are considered the backend. The header value `DENY` applied to the frontend configurations overrides the same header with the value `SAMEORIGIN` that is set in the backend section:
+
+[source,text]
+----
+frontend public
+  http-response set-header X-Frame-Options 'DENY'
+
+frontend fe_sni
+  http-response set-header X-Frame-Options 'DENY'
+
+frontend fe_no_sni
+  http-response set-header X-Frame-Options 'DENY'
+
+backend be_secure:openshift-monitoring:alertmanager-main
+  http-response set-header X-Frame-Options 'SAMEORIGIN'
+----
+
+Additionally, any actions defined in either the Ingress Controller or a route object override values set using route annotations.
+
+== Special case headers
+
+The following headers are either prevented entirely from being set or deleted, or allowed under specific circumstances:
+
+.Header configuration options
+[cols="1,1,1,80,15a",options="header"]
+|===
+|Header name |Configurable using IngressController `spec.httpHeaders.actions` |Configurable using Route `spec.httpHeaders.actions` |Reasoning for disallowment in CRs |Configurable using another method
+
+|Proxy
+|No
+|No
+|The `proxy` HTTP request header can be used to exploit vulnerable CGI applications by injecting the header value into the `HTTP_PROXY` environment variable. The `proxy` HTTP request header is also non-standard, and prone to error during configuration.
+|No
+
+|Host
+|No
+|Yes
+|When the `host` HTTP request header is set using the `IngressController` CR, it can cause HAProxy to fail when looking up the correct route.
+|No
+
+|Strict-Transport-Security
+|No
+|No
+|The `strict-transport-security` HTTP response header is already handled using route annotations, and does not need a separate implementation.
+|Yes: the `haproxy.router.openshift.io/hsts_header` route annotation
+
+|X-Forwarded-For
+|Yes
+|Yes
+|N/A
+|Yes:
+
+* the `haproxy.router.openshift.io/set-forwarded-headers` route annotation
+* the `spec.httpHeaders.forwardedHeaderPolicy` Ingress Controller field
+
+|X-SSL
+|Yes
+|Yes
+|N/A
```

Comment:
Same here, this row provides no information

Reply:
removed